Sema: Add a fix-it for integer literal to option set conversion.

5at5ish opened the pull request:
If expression type is `IntegerLiteralExpr` and value is equal to `0`, 
suggest `[]` as a replacement.

Resolves [SR-6716](https://bugs.swift.org/browse/SR-6716).

jrose-apple reviewed (commented):
This looks great! One formatting comment, but the implementation itself looks fine.

5at5ish committed:
Sema: Add a fix-it for integer literal to option set conversion.

If expression type is IntegerLiteralExpr and value is equal to '0',
suggest '[]' as a replacement.

Resolves: rdar://problem/20704317.

5at5ish commented on lib/Sema/CSDiag.cpp:
```diff
@@ -2533,6 +2539,12 @@ static bool tryRawRepresentableFixIts(InFlightDiagnostic &diag,
+        cast<IntegerLiteralExpr>(expr)->getDigitsText() == "0") {
+      diag.fixItReplace(expr->getSourceRange(), "[]");
+      return true;
+    }
```
@jrose-apple Fixed.

xedin commented:
@swift-ci please smoke test

jrose-apple commented:
Merged! Please move the JIRA bug to Resolved. Thanks, Timur, and welcome!

5at5ish commented:
Thanks, Jordan!
